docs/manual/java/guide/broker/code/docs/javadsl/mb/AnotherServiceImpl.java

```diff
@@ -55,4 +62,60 @@ private void subscribeWithMetadata() {
         //#subscribe-to-topic-with-metadata
 
     }
+
+    private void skipMessages() {
+        //#subscribe-to-topic-skip-messages
+        helloService.greetingsTopic()
+            .subscribe()
+            .atLeastOnce(Flow.fromFunction((GreetingMessage message) -> {
+                if (message.getMessage().equals("Kia ora")) {
+                    return doSomethingWithTheMessage(message);
+                } else {
+                    // Skip all messages that do not start with "Hello"
+                    return Done.getInstance();
+                }
+            }));
+        //#subscribe-to-topic-skip-messages
+    }
+
+    //#subscribe-to-topic-partition-messages
+    private void subscribeToHelloMessages() {
```

Comment:
Maybe adding an ASCII-art diagram of the complete flow would help explain what's going on.

```
+-----------------------------------+
|                                   |
|            ->  logic ->           |
|-> split <               > fanIn ->|
|            -> ignore ->           |
|                                   |
+-----------------------------------+
```

ASCII-art (or any form of art) is really not my thing.